packages/deploy-script-support/src/coreProposalBehavior.js

```diff
@@ -24,21 +31,21 @@ export const permits = {
  * definitions.
  *
  * @param {object} opts
```

Comment:
All of #8528 is cherry-picked.

https://github.com/Agoric/agoric-sdk/pull/8712/commits/5753e80ef91051d20a7d693368849fe5234e5be0 of #8712 is omitted as it depends on https://github.com/Agoric/agoric-sdk/pull/8153/commits/b735591b55199caee53f8fb1b046918aba0c23ea (of #8153)

I would love to bring both these in, but I start having tsc errors in inter-protocol's `add-collateral-core.js` in that case. I actually do not understand how the errors don't happen in `master`. When I look at the `wrapInstall` in the `default` export of  `add-collateral-core.js` in master (moved to the `builders` package), all types seem to be `any`. I suspect `master` doesn't have type errors simply because we lost type coverage somehow there.

@turadg any idea?

Reply:
Getting type errors in `init-core.js` as well.

Reply:
Confirmed. Fix in https://github.com/Agoric/agoric-sdk/pull/8778